protocol/contracts/v2/access/TempleElevatedAccess.sol

```diff
@@ -31,55 +31,89 @@ abstract contract TempleElevatedAccess is ITempleElevatedAccess {
+        emit NewRescuerAccepted(rescuer, msg.sender);
+        rescuer = msg.sender;
+        delete _proposedNewRescuer;
     }
 
     /**
-     * @notice Grant `allowedCaller` the rights to call the function determined by the selector `fnSelector`
+     * @notice Proposes a new Executor.
+     * Can only be called by the current executor or rescuer (if in resuce mode)
+     */
+    function proposeNewExecutor(address account) external override onlyElevatedAccess {
+        if (account == address(0)) revert CommonEventsAndErrors.InvalidAddress();
+        emit NewExecutorProposed(executor, _proposedNewExecutor, account);
+        _proposedNewExecutor = account;
+    }
+
+    /**
+     * @notice Caller accepts the role as new Executor.
+     * Can only be called by the proposed executor
+     */
+    function acceptExecutor() external override {
+        if (msg.sender != _proposedNewExecutor) revert CommonEventsAndErrors.InvalidAccess();
+        emit NewExecutorAccepted(executor, msg.sender);
+        executor = msg.sender;
+        delete _proposedNewExecutor;
+    }
+
+    /**
+     * @notice Grant `allowedCaller` the rights to call the function selectors in the access list.
      * @dev fnSelector == bytes4(keccak256("fn(argType1,argType2,...)"))
      */
-    function setExplicitAccess(address allowedCaller, bytes4 fnSelector, bool value) external override onlyElevatedAccess {
-        if (allowedCaller == address(0)) revert CommonEventsAndErrors.InvalidAddress(allowedCaller);
-        if (fnSelector == bytes4(0)) revert CommonEventsAndErrors.InvalidParam();
-        emit ExplicitAccessSet(allowedCaller, fnSelector, value);
-        explicitFunctionAccess[allowedCaller][fnSelector] = value;
+    function setExplicitAccess(address allowedCaller, ExplicitAccess[] calldata access) external override onlyElevatedAccess {
+        if (allowedCaller == address(0)) revert CommonEventsAndErrors.InvalidAddress();
+        uint256 _length = access.length;
+        ExplicitAccess memory _access;
+        for (uint256 i; i < _length; ++i) {
+            _access = access[i];
+            if (_access.fnSelector == bytes4(0)) revert CommonEventsAndErrors.InvalidParam();
+            emit ExplicitAccessSet(allowedCaller, _access.fnSelector, _access.allowed);
+            explicitFunctionAccess[allowedCaller][_access.fnSelector] = _access.allowed;
+        }
     }
 
     function isElevatedAccess(address caller, bytes4 fnSelector) internal view returns (bool) {
         if (inRescueMode) {
             // If we're in rescue mode, then only the rescuers can call
-            return rescuers[caller];
-        } else if (executors[caller] || explicitFunctionAccess[caller][fnSelector]) {
+            return caller == rescuer;
+        } else if (caller == executor || explicitFunctionAccess[caller][fnSelector]) {
             // If we're not in rescue mode, the executors can call all functions
```

Comment:
nit: comment fix. the executor

Reply:
Fixed in https://github.com/TempleDAO/temple/pull/814/commits/811333f4701aee0aec587d057b77c6935a1a7e61